src/main/scripts/jenkins

```diff
@@ -2,33 +2,66 @@
 
 set -e
 
-nprocs="$(grep -c '^processor' /proc/cpuinfo)"
+notif() {
+    echo -e "\n$@" \
+            "\n================================================================================" \
+            "\n" >&2
+}
 
+NPROCS="${NPROCS:-$(grep -c '^processor' /proc/cpuinfo)}"
+WORKSPACE="${WORKSPACE:-$(pwd)}"
+
+K_EXERCISES_REPO="${K_EXERCISES_REPO:-git@github.com:kframework/k-exercises.git}"
+RVMATCH_REPO="${RVMATCH_REPO:-git@github.com:runtimeverification/rv-match.git}"
+
+KSERVER_SOCKET="$WORKSPACE/kserver"
+KSERVER_LOG="$WORKSPACE/kserver.log"
+export KSERVER_SOCKET
+
+notif "CLEANING REPO"
+mvn clean
+git clean -dffx
+
+notif "SETTING UP OPAM/PERL ENVIRONMENTS"
 eval `opam config env`
 eval `perl -I$HOME/perl5/lib/perl5 -Mlocal::lib`
 
-echo "=== RUNNING LOCAL TESTS ==="
-cd ${WORKSPACE}/k-pr
+notif "CLONING SUBMODULES"
 git submodule update --init
+
+notif "BUILDING/RUNNING LOCAL TESTS"
 mvn verify -U
 
-echo "=== RUNNING K EXERCISES ==="
-cd ${WORKSPACE}/k-pr
-rm -rf k-exercises
-git clone 'git@github.com:kframework/k-exercises.git'
-cd k-exercises/tutorial
-export KSERVER_SOCKET=$(pwd)/kserver
-../../k-distribution/target/release/k/bin/spawn-kserver kserver.log
+notif "STARTING KSERVER"
+./k-distribution/target/release/k/bin/spawn-kserver "$KSERVER_LOG"
 sleep 5
-make -j"$nprocs"
-../../k-distribution/target/release/k/bin/stop-kserver || true
 
-echo "=== RUNNING RV-MATCH TESTS ==="
-cd ${WORKSPACE}
+notif "CLONING K EXERCISES"
+rm -rf k-exercises
+git clone "$K_EXERCISES_REPO" k-exercises
+
+notif "RUNNING K EXERCISES"
+exit_status='0'
+(   cd k-exercises/tutorial
+    make -j"$NPROCS"
+) || exit_status="$?"
+
+notif "KILLING KSERVER"
+./k-distribution/target/release/k/bin/stop-kserver || true
+[[ "$exit_status" == '0' ]] || exit "$exit_status"
+
+notif "CLONING/PREPARING RV-MATCH"
 rm -rf rv-match
-git clone git@github.com:runtimeverification/rv-match.git
-cd rv-match
-git submodule update --init -- c-semantics
-rm -rf k
-cp -r ${WORKSPACE}/k-pr k
-mvn verify -U -DskipKTest -DbuildProfile=x86_64-gcc-glibc
+git clone "$RVMATCH_REPO" rv-match
+(   cd rv-match
+    git submodule update --init -- c-semantics k
+    cd k
+    git fetch ../../
+    git checkout FETCH_HEAD
```

Comment:
What is this? I liked it better when we were just copying the directory into place. This feels error prone to me.

Reply:
The reason to do this is because we want the rv-match tests to be runnable from directly in the K directory (without having to move to a temporary directory or work from one directory up like we were before).

The `git fetch ../../` will fetch the current `HEAD` of the repository at `../../`, which is the current checkout of K. `git fetch` with an anonymous remote (here `../../`) updates `FETCH_HEAD` to whatever the remote `HEAD` is. Nothing too fancy here.

Benefits of this approach:

-   Jenkins script is directly runnable from the K repo directory (trying to use `cp` to do this would result in recursively copying the rv-match repo into itself).
-   We are sure that the rv-match submodule is updated to exactly the version of K that is being tested, because we fetch it directly.